### ISSUE TITLE
Promoting learningOrchestra- Ideas

### ISSUE THREAD

**Issue by navendu-pottekkat** [riibeirogabriel/learningOrchestra#79]
These are some ideas I have to promote this project. 

We can share ideas here and grow the community around this project.

**Sharing in the following platforms**

https://news.ycombinator.com/show:

https://www.reddit.com/

https://www.producthunt.com/

https://dev.to

https://twitter.com/home

We can discuss a strategy and decide the posts on these platforms and then plan a release.

**Comment by navendu-pottekkat** [comment]
I have tried these platforms and actually made it to trending repos in GitHub. After that we would get daily visits and users and contributors. Then we can make this project even better!

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat good! I will make a post in these platforms.

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat we need stars to be a trending repo?

**Comment by navendu-pottekkat** [comment]
> @navendu-pottekkat good! I will make a post in these platforms.

It should be planned and timing is also key. It is all about driving traffic initially. Do let me know how I can help.

Also I would suggest using the banner image in the README as the repo preview image

**Comment by riibeirogabriel** [comment]
How we plan the posts to make a right time? There are good times to publicate?

**Comment by navendu-pottekkat** [comment]
We could spread our posts across different timezones.

We could make a post that would hit the US in the morning.

Then we could also make a post that would hit the eastern parts too.

This is very useful because people check these stuff at particular times and we need to reach them during those times.

It is all about prioritizing. And the posts should be well planned and they would be different for each platform.

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat good, how we can split the tasks?

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat again, thanks! you are awesome!

**Comment by navendu-pottekkat** [comment]
You could start by creating a ProductHunt account, they have some validation period. We will first try to get featured in ProductHunt. I would suggest you to go through the products in the site to get an idea and we can think of how we can make a post. I will also think of the post and we can collaborate.

**Comment by riibeirogabriel** [comment]
Okay, i wiil make it now.

**Comment by navendu-pottekkat** [comment]
@riibeirogabriel When should we release?

**Comment by riibeirogabriel** [comment]
You have more knowledge in promoting, i think which we can start spreading in any time, what do you think?

**Comment by navendu-pottekkat** [comment]
We can first make a soft release on Reddit, Twitter and y combinator. Maybe we can write about the release as an article in Dev.to and Towards Data Science. What do you think?

**Comment by riibeirogabriel** [comment]
You have more

> We can first make a soft release on Reddit, Twitter and y combinator. Maybe we can write about the release as an article in Dev.to and Towards Data Science. What do you think?

Sounds very good, i will talk with my advisors about this article, they can help in this part, and we can stay with the promotion part, what do you think?

**Comment by navendu-pottekkat** [comment]
Okay cool! I would suggest you to post on relevant subreddits the link to the repo. With the header text. Also we can tweet with relevant hashtags. I will reach out to some ML open source supporters on LinkedIn and try to get them to give a shout out.

You can start at maybe 9 AM LA Time which is like 2 hours from now on Reddit and Twitter and Y combinator and if you are in groups in LinkedIn, then there too.

I will also tweet and post on Linkedin and in groups

**Comment by navendu-pottekkat** [comment]
From my experience, most of the traffic generally comes from y combinator, reddit and twitter. I will share the post content in just a minute

**Comment by riibeirogabriel** [comment]
Great! i will make this, do you know good subreddits to make this posts?

**Comment by riibeirogabriel** [comment]
> From my experience, most of the traffic generally comes from y combinator, reddit and twitter. I will share the post content in just a minute

Then i will share this in 9 am from LA?

**Comment by navendu-pottekkat** [comment]
Ya 9 AM LA time is good right? It is still not late in Europe and the morning time in America so we could hit some traffic

**Comment by navendu-pottekkat** [comment]
What do you think about this for Reddit?

learningOrchestra: Easily combine your various library and infrastructure needs for data science projects

Add the link and then in the comments, add a comment similar to our README file that is short but explains the purpose of learningOrchestra.

You can post it on relevant subreddits like:

r/MachineLearning

**Comment by navendu-pottekkat** [comment]
@LaChapeliere You did the README, do you have any suggestions for the posts? I think it would be best if we plan it for a day and the post right.

And after we have decided on the posts, we can release it tomorrow?

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat I did talk with my advisors and  they think which now the learningOrchestra does'nt a stable version, with alll features planned in monograph, then maybe we can wait a time till we get a version with all planned features and then start the promotion, sorry if i disappoint you, but yet there are some features to be implemented.

**Comment by navendu-pottekkat** [comment]
Yes, that is why I talked about the soft release. Maybe if we get more developers in they can help accelerate the development.

**Comment by riibeirogabriel** [comment]
Yep, you is right, they are according with your suggestions about a soft release to get more developers, i will defend my monograph in this week, they want a time to discuss possible alternativies in learningOrchestra features from evaluator feedback, to they plan this soft release, do you can wait one week to start this release?

**Comment by riibeirogabriel** [comment]
> What do you think about this for Reddit?
> 
> learningOrchestra: Easily combine your various library and infrastructure needs for data science projects
> 
> Add the link and then in the comments, add a comment similar to our README file that is short but explains the purpose of learningOrchestra.
> 
> You can post it on relevant subreddits like:
> 
> r/MachineLearning

sounds very good, maybe we can edit adding the beta version and which we need more developers.

**Comment by LaChapeliere** [comment]
I'm jumping on the bandwagon, from my outsider point of view it sounds like a good idea to wait until the feedback from @riibeirogabriel 's defense before promoting the project. If you give me a date, I can also plan so I'm done going through the doc, that way you'll maximise your chances of devs contributing or using the project.

**Comment by navendu-pottekkat** [comment]
Sure that sounds good

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat after the monograph defense, we decide make an api refactoring, joining all requests of each microservice to one common endpoint, and also improving the semantic names in requests, then the user will can use the software more easily, we can start the promotion after this refactor, i will share with you the sketch when is done and you can suggest improviments before i start the implementation.

**Comment by navendu-pottekkat** [comment]
Sure! Let me know and I will be happy to help

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat please take a look in the sketch to new learning orchestra api, we will add an api gateway to abstract the microservices to user, then the user only need know the paths to each feature and make the requests, then after we add the api gateway the user don't need know about microservices. Please give a feedback if sounds good to you, suggestions are welcome! after we implement the api gateway we can begin the learning orchestra promotion, i will add issues about the development to possible new contributors.
https://drive.google.com/file/d/1p0AG2Kb1UEOhTEvxV1Mb1xML3p42HZqd/view?usp=sharing

**Comment by riibeirogabriel** [comment]
Please ignore the /Any JS path in /Api, it is about other project which will be added in learning orchestra infrastructure, separate from learning orchestra functionalities.

**Comment by navendu-pottekkat** [comment]
@riibeirogabriel Looks good to me. Looking forward to the release!

**Comment by hiperbolt** [comment]
I like the API gateway. Will the client(in the other repo) work with this gateway?

**Comment by riibeirogabriel** [comment]
@hiperbolt yes, we will make a refactoring in client when the API gateway is done, we did think use this api gateway https://www.krakend.io/.

**Comment by riibeirogabriel** [comment]
@navendu-pottekkat When the API gateway is done with the refactored requests, we will able to begin the promotion, i will notify you when is done, probably in one or two weeks, if developers come now, with this refactoring in progress, can be hard to understand the software.
